File: svc-event/src/main/java/com/beancounter/event/service/TaxService.java
package com.beancounter.event.service;

import java.math.BigDecimal;
import java.util.HashMap;
import java.util.Map;
import org.springframework.stereotype.Service;

@Service
public class TaxService {
  private final Map<String, BigDecimal> rates = new HashMap<>();

  public TaxService() {
    rates.put("USD", new BigDecimal(".30"));
  }

  public BigDecimal getRate(String code) {
    BigDecimal rate = rates.get(code);
    if (rate == null) {
      return BigDecimal.ZERO;
    }
    return rate;
  }
}

File: svc-event/src/main/java/com/beancounter/event/service/TaxService.kt
package com.beancounter.event.service

import org.springframework.stereotype.Service
import java.math.BigDecimal
import java.util.HashMap

@Service
class TaxService {
    private val rates: MutableMap<String, BigDecimal> = HashMap()
    fun getRate(code: String?): BigDecimal {
        return rates[code] ?: return BigDecimal.ZERO
    }

    init {
        rates["USD"] = BigDecimal(".30")
    }
}
